src/test/ui/associated-types/defaults-specialization.stderr

```diff
@@ -0,0 +1,27 @@
+error[E0053]: method `make` has an incompatible type for trait
+  --> $DIR/defaults-specialization.rs:15:18
+   |
+LL |     fn make() -> Self::Ty;
+   |                  -------- type in trait
+...
+LL |     fn make() -> u8 { 0 }
+   |                  ^^ expected associated type, found u8
+   |
+   = note: expected type `fn() -> <A<T> as Tr>::Ty`
+              found type `fn() -> u8`
```

Comment:
I wonder if it would be feasible in not-too-hacky a manner to tell the user that defaults are the *cause* of the failure (both when using specialization and not). Otherwise the user might be reasonably confused as to why they don't get to assume the default since they are unlikely to be aware of the type theoretical reasons.

Reply:
This could also be left as future work; if you prefer that, please leave a checkbox in the tracking issue :)

Also cc @estebank

Reply:
Yeah I'll do this. This is challenging enough as is ;)

Reply:
For that future work... One thing you might try in the unhappy path is to check whether there is a default that would have allowed it to succeed if not for our rules.